src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let between = separator;
+
+    if (separator === undefined) {
+      between = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === null || this[i] === undefined) & i + 1 !== this.length) {
+        result += `${between}`;
+        continue;
+      }
+
+      if (i + 1 !== this.length) {
+        result += `${this[i]}${between}`;
+        continue;
+      }
+
+      if (this[this.length - 1] !== undefined) {
+        result += this[i];
```

Comment:
I would like to offer you such simplification for this part of code:
1) if it is not the first element, add separator to result
2) if it is null or undefined, just skip this iteration
3) else add the element to the result